go-controller/pkg/ovn/endpoints.go

```diff
@@ -64,43 +64,34 @@ func (ovn *Controller) AddEndpoints(ep *kapi.Endpoints) error {
 
 	protoPortMap := ovn.getLbEndpoints(ep)
 	klog.V(5).Infof("Matching service %s ports: %v", svc.Name, svc.Spec.Ports)
-	for proto, portMap := range protoPortMap {
-		for svcPortName, lbEps := range portMap {
-			ips := lbEps.IPs
-			targetPort := lbEps.Port
-			for _, svcPort := range svc.Spec.Ports {
-				if svcPort.Protocol == proto && svcPort.Name == svcPortName {
-					if !ovn.SCTPSupport && proto == kapi.ProtocolSCTP {
-						klog.Errorf("Rejecting endpoint creation for unsupported SCTP protocol: %s, %s", ep.Namespace, ep.Name)
+	for _, svcPort := range svc.Spec.Ports {
+		if portMap, ok := protoPortMap[svcPort.Protocol]; ok {
+			if lbEps, ok := portMap[svcPort.Name]; ok {
```

Comment:
It's considered bad style to write an `if` such that the entire rest of the enclosing block ends up indented an extra level, in the body of the if; you should flip the sense of the condition and use `continue` to end the `if` statement early, so that the rest of the `range svc.Spec.Ports` loop doesn't end up intended two extra levels for no real reason:

    portMap, ok := protoPortMap[svcPort.Protocol]
    if !ok {
            continue
    }
    lbEps, ok := portMap[svcPort.Name]
    if !ok {
            continue
    }
    if !ovn.SCTPSupport ...

However, the `portMap` check is wrong anyway; `protoPortMap` always contains entries for TCP, UDP, and SCTP, regardless of whether the service defines any ports for them, so `ok` will always be `true`. So you can just simplify all this to

    lbEps := protoPortMap[svcPort.Protocol][svcPort.Name]
    if lbEps == nil {
            continue
    }
    ...

Reply:
Understood, will fix.